client/src/components/Graph/D3Plot.vue

```diff
@@ -0,0 +1,307 @@
+      const yAxisLatexSvg = MathJax.tex2svg(yLabel, yAxisLatexOptions)
+        .firstChild;
+
+      yAxisNode.append(yAxisLatexSvg);
+      // yAxisNode.setAttribute('transform-origin', 'center center');
+
+      // Rotation happens at the pivot point of the top left corner by default
+      yAxisNode.setAttribute('transform', 'rotate(-90)');
+
+      /* Get the yAxisLatexSvg back to the top left of the SVG after rotation
+      then calculate the half way point to line it up at */
+      const yAxisLatexBBox = yAxisLatexSvg.getBoundingClientRect();
+      yAxisContainer.setAttribute('y', yAxisLatexBBox.height
+        + (h / 2)
+        - (yAxisLatexBBox.height / 2));
+      yAxisContainer.setAttribute('overflow', 'visible');
+
+      return {
+        yLabelWidth: yAxisLatexBBox.width,
+        xLabelHeight: xAxisLatexSvg.getBoundingClientRect().height,
+      };
+    },
+    /**
+     * Creates the legend for the plot.
+     *
+     * @param {} svg the d3 svg representing the plot svg
+     * @param {} colorGen the d3 color scale for the plot
+     * @returns {Number} the width of the Legend
+     */
+    generateLegend(svg, colorGen) {
+      const dataSetNames = Object.keys(this.d3PlotData.plot_data);
+      const w = svg.node().getBoundingClientRect().width;
+      const h = svg.node().getBoundingClientRect().height;
+      svg.append('g')
+        .attr('id', 'legendColor')
+        .attr('transform', `translate(${w / 2},20)`);
+      const colorLegend = legendColor()
+        .scale(colorGen)
+        .orient('veritcal')
+        .labels(({
+          i,
+        }) => dataSetNames[i])
+        .labelWrap(150)
+        .on('cellclick', (event) => {
+          const tspanNode = event.target.parentNode.querySelector('tspan');
+          const nodeText = tspanNode.textContent;
+          const cellIndex = dataSetNames.indexOf(nodeText);
+          const classString = `.line-${cellIndex}`;
+          // is the element currently visible ?
+          const currentOpacity = d3.selectAll(classString).style('opacity');
+          // Change the opacity: from 0 to 1 or from 1 to 0
+          if (currentOpacity === '1') {
+            d3.selectAll(classString).style('opacity', '0');
+            tspanNode.setAttribute('text-decoration', 'line-through');
+            tspanNode.setAttribute('fill', 'lightgray');
+          } else {
+            d3.selectAll(classString).style('opacity', '1');
+            tspanNode.removeAttribute('text-decoration');
+            tspanNode.removeAttribute('fill');
+          }
+        });
+
+      svg.select('#legendColor')
+        .call(colorLegend);
+      const legendNode = svg.select('#legendColor').node();
+      const legendBBox = legendNode.getBoundingClientRect();
+      const targetX = w - legendBBox.width;
+      const targetY = (h / 2) - (legendBBox.height / 2);
+      legendNode.setAttribute('transform', `translate(${targetX},${targetY})`);
+      return legendBBox.width;
+    },
+  },
```

Comment:
Literally all of this logic would be be perfectly fine to put in a pure JS class to keep it separate from the data/component itself. Would prevent you needing the thin wrapper for the vue component you have at index.js.

Reply:
Also just while we're at it, it might make more sense to have the 'getPlotTypes' call in the store and call it ONCE on store initialization (and save it in store as plotTypes). Wouldn't be a big change but would clean up index.js substantially.

Reply:
This was done as requested.